Mathlib/Analysis/InnerProductSpace/JointEigenspace.lean

```diff
@@ -28,85 +34,108 @@ and a proof obligation that the basis vectors are eigenvectors.
 
 ## Tags
 
-self-adjoint operator, simultaneous eigenspaces, joint eigenspaces
+symmetric operator, simultaneous eigenspaces, joint eigenspaces
 
 -/
 
-variable {𝕜 E : Type*} [RCLike 𝕜]
-variable [NormedAddCommGroup E] [InnerProductSpace 𝕜 E]
-
 open Module.End
 
 namespace LinearMap
 
 namespace IsSymmetric
 
-section Pair
+variable {𝕜 E n m : Type*}
+
+open Submodule
 
-variable {α : 𝕜} {A B : E →ₗ[𝕜] E}
+section RCLike
 
-/--If a pair of operators commute, then the eigenspaces of one are invariant under the other.-/
-theorem eigenspace_invariant_of_commute
-    (hAB : A ∘ₗ B = B ∘ₗ A) (α : 𝕜) : ∀ v ∈ (eigenspace A α), (B v ∈ eigenspace A α) := by
-  intro v hv
-  rw [eigenspace_def, mem_ker, sub_apply, smul_apply, one_apply, ← comp_apply A B v, hAB,
-    comp_apply B A v, ← map_smul, ← map_sub, hv, map_zero] at *
+variable [RCLike 𝕜] [NormedAddCommGroup E] [InnerProductSpace 𝕜 E]
+variable {α : 𝕜} {A B : E →ₗ[𝕜] E} {T : n → E →ₗ[𝕜] E}
```

Comment:
```suggestion
variable {α : 𝕜} {A B : E →ₗ[𝕜] E} {T : n → Module.End 𝕜 E}
```
This will allow dot notation to access the lemmas in the `Module.End` namespace (and is required for my final comment to work).